### PR TITLE
docs (bodiless-documentation): Add content editor docs to YouTube and iframe components

### PR DESCRIPTION
**Pull request by malachye**
## Changes
Edit and reformat documentation for the following components:
- Iframe
- YouTube
- Embed 

## Related Issues

#627 
https://jira.jnj.com/browse/AESQ-3638

### CODE REVIEW

File: packages/bodiless-components/doc/YouTube.md [hunk 1]
```diff
@@ -1,16 +1,45 @@
 # YouTube Component
```
Comment (hvanyo) [inline packages/bodiless-components/doc/YouTube.md:1]: https://edit.t2kekqmnku-jvaff4bu65vgm.us-2.platformsh.site/___docs/#/Components/YouTube -- the left hand side of H1 is getting a space in You Tube.  I don't think there is anything you can do other than rename it to Youtube.md and then you Components | Youtube which is also wrong.  You can try put the file and name into https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-documentation/src/defaultToc.ts and see if this help.       Components: {
      'YouTube.md': 'YouTube',

File: packages/bodiless-components/doc/YouTube.md [hunk 2]
```diff
@@ -1,16 +1,45 @@
...
 
-  The YouTube Component can be used to place a YouTube video on a page via the BodilessJS edit interface.
+The YouTube Component can be used to place a YouTube video on a page via the
```
Comment (hvanyo) [inline packages/bodiless-components/doc/YouTube.md:3]: Link the Youtube video - to https://www.youtube.com/

File: packages/bodiless-components/doc/YouTube.md [hunk 3]
```diff
@@ -1,16 +1,45 @@
...
+BodilessJS edit interface. The YouTube Component is responsive and features 
+a 16:9 ratio by default. 
 
-  ``` js
+## Content Editor Details
+
+To add a YouTube Component to your site:
+
+1. In Edit Mode, select the YouTube Component from the Component Library.
```
Comment (hvanyo) [inline packages/bodiless-components/doc/YouTube.md:11]: No images still :(

File: packages/bodiless-components/doc/YouTube.md [hunk 4]
```diff
@@ -1,16 +1,45 @@
...
+![](./assets/YoutubeComponentLibrary.jpg)
+
+2. Click on the YouTube Component to show the Configure button in the context menu. 
+![](./assets/YoutubeFlow.jpg)
+
+3. Enter the target YouTube URL and click the checkmark or hit enter. 
+?> Please note the following accepted URL formats:
```
Comment (hvanyo) [inline packages/bodiless-components/doc/YouTube.md:18]: This note is not displaying properly https://edit.t2kekqmnku-jvaff4bu65vgm.us-2.platformsh.site/___docs/#/Components/YouTube

File: packages/bodiless-components/doc/YouTube.md [hunk 5]
```diff
@@ -34,11 +64,75 @@
...
+### Props
 
 | name           | default   | description                                                                         |
 |----------------|-----------|-------------------------------------------------------------------------------------|
 | nodeKey        | undefined | Identifies where the component data will be stored.                                 |
 | playerSettings | undefined | YouTube embed player settings. Check YouTubePlayerSettings type from 'Youtube.tsx'. |
 
 In addition, you can pass `HTMLIFrameElement` props (id, src, width, etc.). Check `HTMLIFrameElement` type for more details.
+
+### Parameters
```
Comment (hvanyo) [inline packages/bodiless-components/doc/YouTube.md:76]:
```suggestion
### `playerSettings` Parameters 
```

File: packages/bodiless-components/doc/YouTube.md [hunk 6]
```diff
@@ -34,11 +64,75 @@
...
 | nodeKey        | undefined | Identifies where the component data will be stored.                                 |
 | playerSettings | undefined | YouTube embed player settings. Check YouTubePlayerSettings type from 'Youtube.tsx'. |
 
 In addition, you can pass `HTMLIFrameElement` props (id, src, width, etc.). Check `HTMLIFrameElement` type for more details.
+
+### Parameters 
+
+The YouTube Component utilizes an API that allows you to change the following
+properties/parameters. You can use defaultPlayerSettings to set parameters that
+are commonly used together. For more details see [YouTube Player API Reference for iframe Embeds](https://developers.google.com/youtube/iframe_api_reference).
```
Comment (hvanyo) [inline packages/bodiless-components/doc/YouTube.md:80]: For more details on each individual parameter see

File: packages/bodiless-components/doc/YouTube.md [hunk 7]
```diff
@@ -34,11 +64,75 @@
...
 In addition, you can pass `HTMLIFrameElement` props (id, src, width, etc.). Check `HTMLIFrameElement` type for more details.
+
+### Parameters 
+
+The YouTube Component utilizes an API that allows you to change the following
+properties/parameters. You can use defaultPlayerSettings to set parameters that
+are commonly used together. For more details see [YouTube Player API Reference for iframe Embeds](https://developers.google.com/youtube/iframe_api_reference). 
+
+
+| Parameters | Default |
```
Comment (hvanyo) [inline packages/bodiless-components/doc/YouTube.md:83]: Suggested Default

File: packages/bodiless-components/doc/YouTube.md [hunk 8]
```diff
@@ -1,16 +1,45 @@
...
+4. Resize or reposition the YouTube component using the Flow Container controls.
+
+
+---
+
+## Site Builder Details
+
+The YouTube Component can be used to place a YouTube video on a page via the 
+BodilessJS edit interface.
+
```
Comment (hvanyo) [inline packages/bodiless-components/doc/YouTube.md:32]: I would add here that Youtube depends on Embed component (link) and needs to have this activated as well on site build.

File: packages/bodiless-components/doc/Iframe.md [hunk 9]
```diff
@@ -1,9 +1,42 @@
 # Iframe Component
 
-  One can use this to compose and place an iframe on the page, that uses
-  the BodilessJS edit interface.
+The Iframe Component can be used to place an iframe on a BodilessJS page.
```
Comment (hvanyo) [inline packages/bodiless-components/doc/Iframe.md:3]: Iframe and iframe in same line -- both right ?

File: packages/bodiless-components/doc/Iframe.md [hunk 10]
```diff
@@ -1,9 +1,42 @@
...
+![](./assets/IframeComponentLibrary.jpg)
+
+2. Click on the iframe component to toggle the configuration button for 
+the iframe.
+![](./assets/IframeConfigButton.jpg)
+
+
+3. Enter the URL you wish to place in the iframe and enter the height at 
+which you would like to display the iframe's content. 
+!> Please note that the originating site for the target URL must be configured
```
Comment (hvanyo) [inline packages/bodiless-components/doc/Iframe.md:21]: This note doesn't appear correctly: https://edit.t2kekqmnku-jvaff4bu65vgm.us-2.platformsh.site/___docs/#/Components/Iframe

File: packages/bodiless-components/doc/Iframe.md [hunk 11]
```diff
@@ -60,8 +99,19 @@
...
 ```
+
+---
+
+## Architectural Details
+
+The Iframe Component renders:
+
+```
+<iframe src="string" height="200"></iframe>
```
Comment (hvanyo) [inline packages/bodiless-components/doc/Iframe.md:115]: height=""

File: packages/bodiless-components/doc/Iframe.md (hunk 9 above)
Comment (malachye) [reply packages/bodiless-components/doc/Iframe.md:3]: my thinking was that when we're talking about the component it is "Iframe Component". If you're just talking about an iframe it is lower case.

File: packages/bodiless-components/doc/YouTube.md (hunk 8 above)
Comment (malachye) [reply packages/bodiless-components/doc/YouTube.md:34]: Added a note

File: packages/bodiless-components/doc/YouTube.md (hunk 1 above)
Comment (malachye) [reply packages/bodiless-components/doc/YouTube.md:1]: should i try to change the TOC first and if that fails change the .md name to Youtube?

File: packages/bodiless-components/doc/Iframe.md (hunk 9 above)
Comment (hvanyo) [reply packages/bodiless-components/doc/Iframe.md:3]: ok  just wondering why.  :)

File: packages/bodiless-components/doc/YouTube.md (hunk 1 above)
Comment (hvanyo) [reply packages/bodiless-components/doc/YouTube.md:1]: Didn't help -- but lets move on.

**Review by hvanyo — DISMISSED** [review]

**Review by kbray0009 — APPROVED** [review]

File: packages/bodiless-documentation/src/defaultToc.ts [hunk 12]
```diff
@@ -19,7 +19,9 @@ const toc = {
   },
   ContentEditorUserGuide: {},
   Design: {},
-  Components: {},
+  Components: {
+     'YouTube.md': 'YouTube',
```
Comment (dewen) [inline packages/bodiless-documentation/src/defaultToc.ts:23]: There's an extra space before first single quote, which breaks the lint.

**Review by hvanyo — APPROVED** [review]

**Review by kbray0009 — APPROVED** [review]

**Pull request merged**